### ISSUE TITLE
The comparing with memory by 1080 ti and 3090

### ISSUE THREAD

**Issue by leeivan1007** [tensorflow/tensorflow#47358]
- Unet by tensorflow.keras (not include tf.compat.v1 code)
- Both of these are ubuntu18
- 2.4.0 (A computer) 2.0.0 (B computer)
- Both of these are python 3.6.12
- 11.1/8.0.5 (A computer) 10.0/7.6.5 (B computer)
- 3090 24265MiB (A computer) 1080 ti 11178MiB (B computer)
- Driver version 460.39

### current behavior

I ran the code using Unet to do segmentation project.
I found the using of memory is very diffecent.
It is the setting of model when I had trained the model:

batch_size = 20
epochs = 2

When I had trained the model, I inspect the htop and nvidia-smi:

The using resource by 3090 (A computer)
cpu memory 4.4G -> 12G
gpu memory 457MiB -> 18307MiB 

The using resource by 1080 ti: (B computer)
cpu memory 0.79G  -> 3.4G
gpu memory 0MiB -> 10867MiB

### expected behavior

I expected 3090(tf2.4) should be faster than 1080(tf2.0).

Thank you to read this issue!

**Comment by ravikyram** [comment]
@leeivan1007 

Please, share the simple standalone code to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by leeivan1007** [comment]
Sorry. I have to discuss with my boss if I share the code. The Unet had changed to some projects...
But I can test with official code:

### Classification on mnist / tf2.4 / 3090

```
import tensorflow as tf

physical_devices = tf.config.list_physical_devices('GPU')
tf.config.experimental.set_memory_growth(physical_devices[0], True)

mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10)
])

loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
model.compile(optimizer='adam',
              loss=loss_fn,
              metrics=['accuracy'])

model.fit(x_train, y_train, epochs=5)
```
These are utilize of memory:
1. 6.5G -> 8.33G (memory) 
2. 549Mib -> 1218MiB (GPU memory)


### Classification on mnist / tf2.0 / 1080

```
import tensorflow as tf
import os

os.environ["CUDA_VISIBLE_DEVICES"] = 0

config = tf.compat.v1.ConfigProto()
config.gpu_options.allow_growth = True
session = tf.compat.v1.Session(config=config)

mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10)
])

loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)

model.compile(optimizer='adam',
              loss=loss_fn,
              metrics=['accuracy'])
model.fit(x_train, y_train, epochs=10)
``
It is an only different that setting of memory growth.

These are utilize of memory:
1.0.8G -> 2.18G 
2.0Mib -> 283MiB

**Comment by ravikyram** [comment]
@leeivan1007 

Can you please test the both using TF version 2.4 and see if the issue still persists. Thanks!

**Comment by memr5** [comment]
Het @leeivan1007, 

I am having the same issue. Did you find the reason behind this issue?

similar threads: 
* https://github.com/NVIDIA/tensorflow/issues/23
* https://forums.developer.nvidia.com/t/new-tensorrt-model-occupying-more-gpu-memory-as-compared-to-older-version/186373
